Tests/MongoSwiftTests/MongoClientTests.swift

```diff
@@ -74,6 +74,38 @@ final class MongoClientTests: MongoSwiftTestCase {
         }
     }
 
+    func testListDatabasesComment() throws {
```

Comment:
Hmm, I think we probably don't want to introduce any more futures-based tests at this point (hence your later projects), so this might be better written using async/await. More generally, maybe we don't want anymore sync tests either, so perhaps it would be good to rewrite the other test you have too. WDYT @kmahar?

As a side note, you should be able to use testing utilities to check if the version matches when using async/await.

Reply:
yeah I think we may as well start writing `async` tests now whenever we can -- we don't even run our tests on Swift versions < 5.5, we only check compilation there, so we aren't going to lose any coverage by writing async/await tests. 
I believe that as of Swift 5.5.3 we can just mark tests `async throws` rather than `throws` to do this (the existing async/await tests had to use a workaround since that wasn't supported at the time we added them; should go back and clean that up at some point)

Reply:
This makes sense. Is the other test I wrote (`testListCollectionsComment()`) not `async`? I was able to use the testing utilities, but I also didn't use `await` keyword. Unsure on what defines an `async` test as opposed to a `sync` test.

Reply:
Updated the `asyncTestUtils` extension to have `serverVersionIsInRange`, created new file for async `MongoDatabaseTests`, made the collection test more async